From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Mon, 8 May 2017 22:58:32 +0200
Subject: [PATCH] {MAYBE UPS} of: fix -Wignored-qualifiers

gcc on ARM reports:

    drivers/of/of_reserved_mem.c: In function '__reserved_mem_init_node':
    drivers/of/of_reserved_mem.c:200:15: error: type qualifiers ignored
    on function return type [-Werror=ignored-qualifiers]
       int const (*initfn)(struct reserved_mem *rmem) = i->data;
                   ^~~~~~
---
 drivers/of/of_reserved_mem.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/of/of_reserved_mem.c b/drivers/of/of_reserved_mem.c
index 4dec07ea510f..c89a64748cdc 100644
--- a/drivers/of/of_reserved_mem.c
+++ b/drivers/of/of_reserved_mem.c
@@ -197,7 +197,7 @@ static int __init __reserved_mem_init_node(struct reserved_mem *rmem)
 	const struct of_device_id *i;
 
 	for (i = __reservedmem_of_table; i < &__rmem_of_table_sentinel; i++) {
-		int const (*initfn)(struct reserved_mem *rmem) = i->data;
+		int (*const initfn)(struct reserved_mem *rmem) = i->data;
 		const char *compat = i->compatible;
 
 		if (!of_flat_dt_is_compatible(rmem->fdt_node, compat))
-- 
